### PR TITLE
Ensure InputReady condition for role/node

### PR DESCRIPTION
**Pull request by rabi**
We don't have a condition to ensure that inputs are ready for the role/node. This adds it and ensures that we check all required keys in the secret.

### CODE REVIEW

**Comment by rabi** [comment]
recheck

**Comment by rabi** [comment]
recheck

File: controllers/common.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	AnsibleSSHPrivateKey = "ssh-privatekey"
+	// AnsibleSSHPublicKey ssh public key
+	AnsibleSSHPublicKey = "ssh-publickey"
+	// AnsibleSSHAuthorizedKeys authorized keys
+	AnsibleSSHAuthorizedKeys = "authorized_keys"
+)
+
+// ensureSSHSecret - ensures that the Secret object exists and the expected fields
+// are in the Secret. It returns a hash of the values of the expected fields.
+func ensureSSHSecret(
```
Comment (bshephar) [inline controllers/common.go:46]: Isn't this essentially just re-implementing the `lib-common` `VerifySecret` function with some condition handling built in?

Were we not able to set these conditions based on the return value of the lib-common function? Just trying to understand the need to reimplement the logic here.

Comment (rabi) [reply controllers/common.go:46]: It's little different, other than checking the secret and fields it updates conditions and does not generate hash which we won't use.

Comment (rabi) [reply controllers/common.go:46]: We can reuse that though. Let me update the PR.

Comment (rabi) [reply controllers/common.go:46]: Hmm.. I remembered the reason I created a different one, to differentiate between status messages (waiting vs error).

**Review by bshephar — APPROVED** [review]
lgtm.

**Pull request merged**